bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -0,0 +1,536 @@
+                new StringType(String.valueOf(resultUrl)));
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.SERVER),
+                new StringType(String.valueOf(server)));
+    }
+
+    /*
+     * Checks to make sure the executable for speedtest is valid
+     */
+    public boolean checkConfig(String execPath) {
+        File file = new File(execPath);
+        if (!checkFileExists(file)) { // Check if entered path exists
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Speedtest executable not found. Please check configuration.");
+            return false;
+        }
+
+        if (!checkFileExecutable(file)) { // Check if speedtest is executable
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Speedtest CLI tool not executable. Please check installation/configuration.");
+            return false;
+        }
+        return true;
+    }
+
+    /*
+     * Executes a given command and returns back the String data of stdout.
+     */
+    private String executeCmd(String commandLine) {
+        int timeOut = 60000;
+        String[] cmdArray;
+        String[] shell;
+        logger.debug("Passing to shell for parsing command.");
+        switch (getOperatingSystemType()) {
+            case WINDOWS:
+                shell = SHELL_WINDOWS;
+                logger.debug("OS: WINDOWS ({})", getOperatingSystemName());
+                cmdArray = createCmdArray(shell, "/c", commandLine);
+                break;
+            case LINUX:
+            case MAC:
+            case SOLARIS:
+                // assume sh is present, should all be POSIX-compliant
+                shell = SHELL_NIX;
+                logger.debug("OS: *NIX ({})", getOperatingSystemName());
+                cmdArray = createCmdArray(shell, "-c", commandLine);
+                break;
+            default:
+                logger.debug("OS: Unknown ({})", getOperatingSystemName());
+                return "";
+        }
+
+        if (cmdArray.length == 0) {
+            logger.debug("Empty command received, not executing");
+            return "";
+        }
+
+        logger.debug("The command to be executed will be '{}'", Arrays.asList(cmdArray));
+
+        Process proc;
+        try {
+            proc = rt.exec(cmdArray);
+        } catch (Exception e) {
+            logger.debug("An exception occurred while executing '{}': '{}'", Arrays.asList(cmdArray), e.getMessage());
+            return "";
+        }
+
+        StringBuilder outputBuilder = new StringBuilder();
+        try (InputStreamReader isr = new InputStreamReader(proc.getInputStream());
+                BufferedReader br = new BufferedReader(isr)) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                outputBuilder.append(line).append("\n");
```

Comment:
The line seperator `\n` is OS specific. Are you sure it will work on all OS-es?
You can use `System.lineSeparator()` to get one depending on the OS.

Reply:
fixed